x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/sample_response/sample_response_logic.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { i18n } from '@kbn/i18n';
+
+import { flashAPIErrors } from '../../../../shared/flash_messages';
+
+import { HttpLogic } from '../../../../shared/http';
+import { EngineLogic } from '../../engine';
+
+import { SampleSearchResponse, ServerFieldResultSettingObject } from '../types';
+
+const NO_RESULTS_MESSAGE = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.resultSettings.sampleResponse.noResultsMessage',
+  { defaultMessage: 'No results.' }
+);
+
+const ERROR_MESSAGE = i18n.translate(
+  'xpack.enterpriseSearch.appSearch.engine.resultSettings.sampleResponse.errorMessage',
+  { defaultMessage: 'An error occured.' }
+);
+
+interface SampleResponseValues {
+  query: string;
+  response: SampleSearchResponse | string | null;
+}
+
+interface SampleResponseActions {
+  queryChanged: (query: string) => { query: string };
+  getSearchResultsSuccess: (response?: SampleSearchResponse) => { response?: SampleSearchResponse };
+  getSearchResultsFailure: (response?: string) => { response?: string };
+  getSearchResults: (
+    query: string,
+    resultFields: ServerFieldResultSettingObject
+  ) => { query: string; resultFields: ServerFieldResultSettingObject };
+}
+
+export const SampleResponseLogic = kea<MakeLogicType<SampleResponseValues, SampleResponseActions>>({
```

Comment:
This is a new logic file, this was inline in the SampleResponse component in the old UI.